src/main/java/xyz/nucleoid/map_templates/BlockBounds.java

```diff
@@ -21,24 +21,17 @@
 
 /**
  * Represents an axis-aligned-bounding-box aligned to the block grid.
- *
+ * <p>
  * This is made up of an inclusive minimum and maximum {@link BlockPos}.
  */
-public final class BlockBounds implements Iterable<BlockPos> {
-    public static final Codec<BlockBounds> CODEC = RecordCodecBuilder.create(instance -> {
-        return instance.group(
-                BlockPos.CODEC.fieldOf("min").forGetter(b -> b.min),
-                BlockPos.CODEC.fieldOf("max").forGetter(b -> b.max)
-        ).apply(instance, BlockBounds::new);
-    });
-
-    private final BlockPos min;
-    private final BlockPos max;
-
-    private BlockBounds(BlockPos min, BlockPos max) {
-        this.min = min;
-        this.max = max;
-    }
+public record BlockBounds(BlockPos min,
+                          BlockPos max) implements Iterable<BlockPos> {
```

Comment:
The newline is unnecessary here.

```suggestion
public record BlockBounds(BlockPos min, BlockPos max) implements Iterable<BlockPos> {
```